core/app/models/spree/promotion_code.rb

```diff
@@ -39,4 +40,8 @@ def usage_limit
   def downcase_value
     self.value = value.downcase
   end
+
+  def remove_excess_whitespace
+    value&.gsub! /\s+/, ''
```

Comment:
Up to you guys regarding the internal whitespace. I will say I have had a real world case where the admin put whitespace in the middle and couldn't see it because of the small font size. The question is how far do we go to make it fool proof since there is always a bigger fool?

We do know whitespace should never be in a code so doesn't seem to hurt to remove it. The person who pays my paycheck is the one who added the whitespace so if you prefer `strip` I'll just have to do a local override.